tenseal/cpp/tensors/ckkstensor.cpp

```diff
@@ -0,0 +1,47 @@
+#include "tenseal/cpp/tensors/ckkstensor.h"
+
+using namespace seal;
+using namespace std;
+
+namespace tenseal {
+
+CKKSTensor::CKKSTensor(const shared_ptr<TenSEALContext>& ctx,
+                       const vector<double>& data, std::optional<double> scale)
```

Comment:
Why isn't data a PlainTensor directly?

Reply:
True that expecting a PlainTensor would make life much easier. I will only encrypt data and use the previously saved strides and shape.